math/src/main/java/com/powsybl/math/timeseries/TimeSeriesTable.java

```diff
@@ -428,17 +480,22 @@ private void computeVariableTimeSeriesPpmcc(double[] r, int timeSeriesNum1, int
 
         Stopwatch stopWatch = Stopwatch.createStarted();
 
-        updateStatistics(version);
+        double[] r = new double[doubleTimeSeriesNames.size()];
 
-        int statisticsIndex1 = getStatisticsIndex(version, timeSeriesNum1);
-        double stdDev1 = stdDevs[statisticsIndex1];
+        statsLock.lock();
+        try {
+            updateStatistics(version);
 
-        double[] r = new double[doubleTimeSeriesNames.size()];
+            int statisticsIndex1 = getStatisticsIndex(version, timeSeriesNum1);
+            double stdDev1 = stdDevs[statisticsIndex1];
```

Comment:
Why do variable names (timeSeriesNum1, statisticsIndex1 and stdDev1) have a suffix? Is it really needed?